### PR TITLE
Sorting

### PR DESCRIPTION
**Pull request by adamklepacz**
Sorting flow looks like this:

1. SortBox component triggers handlers inside the component. 
2. Later on, sorting is handled in App.js component by setting the state of the desired list.
3. To sort inside App.js I'm using function coming from utils folder.

Sorting logic is divided into three conditions:
- sorting by string, 
- sorting by date, 
- sorting by number.

In the database, I'm using UTC string format of the date, not timestamp, because during performing a sorting function it was impossible to distinguish timestamp from eg. '123' number. Btw. every positive integer is recognize as timestamp:

input: `new Date(123)` 
output: `--> Thu Jan 01 1970 01:00:00 GMT+0100 (czas środkowoeuropejski standardowy)`

### CODE REVIEW

File: src/client/components/SortBox.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          <option value="createdAt">Date</option>
+          <option value="name">Name</option>
+        </select>
+        <div className="sort-box__controllers">
+          <button
+            className="sort-box__button"
+            onClick={this.handleOrder}
+            type="button"
+          >
+            <svg
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: move svg to separate files

File: src/client/components/SortBox.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { SortType } from '../common/enums';
+
+class SortBox extends Component {
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: make this component more general as discussed

File: src/client/utils/sortLists.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as moment from 'moment';
+/**
+ *
+ * @param {*} criteria - Criteria argument has to match databse object property. eg. const user = { name: 'Fred'} -> criteria = 'name'
+ */
+const dateFormat = 'YYYY-MM-DDTHH:mm:ss.SSSSZ';
+
+export const sortList = (listName, criteria, order) =>
+  order
```
Comment (marekrozmus) [inline src/client/utils/sortLists.js]: this could be simplified (without code copying) by moving the 'order' check inside the is statements

Comment (adamklepacz) [reply src/client/utils/sortLists.js]: Check if I did it the correct way. If no give me some tip how to make it better

File: src/client/components/App/App.js [hunk 4]
```diff
@@ -45,8 +46,15 @@ class App extends Component {
           />
         )}
         <InputBar />
-        <ProductsContainer products={shoppingList} />
-        <ProductsContainer archived products={archiveList} />
+        <ProductsContainer
+          handleSort={this.handleSort}
```
Comment (marekrozmus) [inline src/client/components/App/App.js]: please remove that

File: src/client/components/ProductsContainer.js [hunk 5]
```diff
@@ -1,21 +1,81 @@
...
+      case OptionType.NAME:
+        result = _sortBy(result, item => item.name.toLowerCase());
+        break;
+      case OptionType.AUTHOR:
+        result = _sortBy(result, [
+          item => item.author.toLowerCase(),
+          item => item.name.toLowerCase()
+        ]);
+        break;
+
```
Comment (marekrozmus) [inline src/client/components/ProductsContainer.js]: remove this new line

File: src/client/components/ProductsContainer.js [hunk 6]
```diff
@@ -1,21 +1,81 @@
...
+const options = [
+  { id: OptionType.AUTHOR, label: 'author' },
+  { id: OptionType.DATE, label: 'date' },
+  { id: OptionType.NAME, label: 'name' }
+];
+
+class ProductsContainer extends Component {
+  state = {
+    sortBy: '',
+    order: false
```
Comment (marekrozmus) [inline src/client/components/ProductsContainer.js]: sortOrder with enum

File: src/client/components/SortBox.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+              {option.label}
+            </option>
+          ))}
+        </select>
+        <div className="sort-box__controllers">
+          <button
+            className={classNames('sort-box__button', {
+              'sort-box__button--obverse': order
+            })}
+            onClick={() =>
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: define onclick method

File: src/client/components/SortBox.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        >
+          {options.map(option => (
+            <option key={option.label} value={option.id}>
+              {option.label}
+            </option>
+          ))}
+        </select>
+        <div className="sort-box__controllers">
+          <button
+            className={classNames('sort-box__button', {
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: please extract classname

File: src/client/components/SortBox.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    const { label, onChange, options } = this.props;
+    const { order, selectedOption } = this.state;
+
+    return (
+      <div className="sort-box">
+        <span className="sort-box__desc">{label}</span>
+        <select
+          className="sort-box__select"
+          onBlur={this.handleSelect}
+          onChange={e =>
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: create event handler

File: src/client/components/SortBox.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+import classNames from 'classnames';
+
+import ArrowIcon from '../assets/images/arrow-up-solid.svg';
+
+class SortBox extends Component {
+  constructor(props) {
+    super(props);
+
+    this.selectField = React.createRef();
+    this.state = {
```
Comment (marekrozmus) [inline src/client/components/SortBox.js]: remove state and use props

File: src/client/components/ProductsContainer.js [hunk 11]
```diff
@@ -1,21 +1,86 @@
...
+  { id: SortOptionType.NAME, label: 'name' }
+];
+
+class ProductsContainer extends Component {
+  state = {
+    sortBy: SortOptionType.NAME,
+    sortOrder: SortOrderType.ASCENDING
+  };
+
+  onSortChange = (sortBy, sortOrder) => {
```
Comment (marekrozmus) [inline src/client/components/ProductsContainer.js]: please remove braces and makeit one line

File: src/client/components/ProductsContainer.js [hunk 12]
```diff
@@ -1,21 +1,86 @@
...
+        result.sort((a, b) => {
+          const dateA = new Date(a.createdAt);
+          const dateB = new Date(b.createdAt);
+          return dateA - dateB;
+        });
+        break;
+      default:
+        break;
+    }
+    return sortOrder === SortOrderType.ASCENDING ? result : result.reverse();
```
Comment (marekrozmus) [inline src/client/components/ProductsContainer.js]: new line before return

**Pull request merged**